packages/snaps-sdk/src/jsx/components/Section.ts

```diff
@@ -0,0 +1,40 @@
+import type { GenericSnapElement, SnapsChildren } from '../component';
+import { createSnapComponent } from '../component';
+
+/**
+ * The props of the {@link Section} component.
+ *
+ * @property children - The children of the section.
+ */
+export type SectionProps = {
+  // We can't use `JSXElement` because it causes a circular reference.
+  children: SnapsChildren<GenericSnapElement>;
```

Comment:
I think we should allow the `direction` and `alignment` props from `Box`

Reply:
`alignment` makes sense but then if we open up `direction` we aren't keeping things vertically aligned (going by the [figma](https://www.figma.com/design/gHWl7G8ih3YXesly2XaA0R/Snaps-DS?node-id=65-1651&t=KTMCUlNQezLPlUyE-0))

Reply:
I talked to @eriknson about it and we just want vertical to be the default, not forced.

Reply:
Same as box really

Reply:
Ah gotcha, also keeping the background color as the default doesn't give you an opportunity to see the visual delineation that was intended with border radius since the background color of the UI is the same. What to do? cc: @eriknson

Reply:
We're changing the background color 👀

Reply:
In another PR

Reply:
Made the changes here [78b4e64](https://github.com/MetaMask/snaps/pull/2672/commits/78b4e64260a9c19ff48b316aca4da61b26ca5d15)